app/controllers/v1/bookings_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module V1
+  class BookingsController < ApplicationController
+    before_action :set_meeting_room, only: [:create]
+    before_action :set_user, only: [:create]
+
+    def create
+      # Check whether Meeting Room is available.
+      return render json: {error: 'Sorry, Meeting Room not available for booking.'}, status: :ok unless @meeting_room.is_available?
+
+      # Book a Meeting Room for a User
+      @booking = @user.bookings.build(
+        meeting_room: @meeting_room,
+        start_time: params[:start_time],
+        end_time: params[:end_time],
+        date: params[:date]
+      )
+      # TODO: before save check whether the time slot is available, if not repond with 'Time slot not Available'
+
+
+      if @booking.save
+        render json: @booking, status: :created
+      else
+        render json: @booking.errors, status: :bad_request
```

Comment:
Why would this return a bad_request ? Would the saving fail because the request is bad?
If the request is bad - then maybe put in some validations?

Reply:
I think this should return 422, Will change it.

Reply:
@glen I googled about it and found that for validation error response we can use 422. Please let me know if we need to use other http code